### PR TITLE
Allowing field mappings

### PR DESCRIPTION
**Pull request by fidesachates**
@sksamuel 

Couldn't find this feature: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-field-mapping.html

This is PR adds the above functionality; it also removes `local` method from `GetMappingRequest` as I couldn't find any usage of it

### CODE REVIEW

File: elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala [hunk 1]
```diff
@@ -74,5 +84,53 @@ class MappingHttpTest extends WordSpec with DockerTests with Matchers with Befor
...
+        getMapping("indexnopropsempty")
+      }.await.result
+
+      val properties = mappings.find(_.index == "indexnopropsempty").get.mappings
+
+      properties shouldBe Map.empty
+    }
+  }
+
+  "specific field mapping get" should {
```
Comment (fidesachates) [inline elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala]: Test cases for when we ask for a specific field in a mapping get request

File: elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala [hunk 2]
```diff
@@ -74,5 +84,53 @@ class MappingHttpTest extends WordSpec with DockerTests with Matchers with Befor
 
       properties shouldBe Map.empty
     }
+
+    "handle properly completely empty mapping" in {
```
Comment (fidesachates) [inline elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/requests/mappings/MappingHttpTest.scala]: New edge case test when the mapping is completely empty

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala [hunk 3]
```diff
@@ -28,12 +29,34 @@ trait MappingHandlers {
       }
     }
 
+    private def getMappings(mappings: Map[String, Any]) = {
+      mappings
+        .getOrElse("properties", getFieldMappings(mappings))
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala]: This is the tricky bit: if the response doesn't have "properties" then it's either 

1. There are no mappings
2. There are mappings and it is a field mapping request (aka when we ask for specific fields in a get mapping request).

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala [hunk 4]
```diff
@@ -28,12 +29,34 @@ trait MappingHandlers {
...
+    }
+
+    private def getFieldMappings(possibleFieldMapping: Map[String, Any]) = {
+      possibleFieldMapping.size match {
+        case 0 => Map.empty
+        case _: Int => possibleFieldMapping.head._2 match {
+          case map: Map[String, Any] =>
+            map.getOrElse("mapping", Map.empty)
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala]: This will match when do a specific field mapping request. I chose "mapping" as a best guess. 
```
{
    "index": {
        "mappings": {
            "a": {
                "full_name": "a",
                "mapping": {
                    "a": {
                        "type": "text",
                        "store": true,
                        "analyzer": "whitespace"
                    }
                }
            }
        }
    }
}
```

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala [hunk 5]
```diff
@@ -28,12 +29,32 @@ trait MappingHandlers {
...
+    private def getMappings(mappings: Map[String, Any]) = {
+      mappings
+        .getOrElse("properties", getFieldMappings(mappings))
+    }
+
+    private def getFieldMappings(possibleFieldMapping: Map[String, Any]) = {
+      possibleFieldMapping.size match {
+        case 0 => Map.empty
+        case _: Int => possibleFieldMapping.head._2 match {
+          case map: Map[String, Any] => map.getOrElse("mapping", Map.empty)
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala]: This will match when do a specific field mapping request. I chose "mapping" as a best guess. 

The response for such a request is below
```
{
    "index": {
        "mappings": {
            "a": {
                "full_name": "a",
                "mapping": {
                    "a": {
                        "type": "text",
                        "store": true,
                        "analyzer": "whitespace"
                    }
                }
            }
        }
    }
}
```

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala [hunk 6]
```diff
@@ -28,12 +29,32 @@ trait MappingHandlers {
...
+          case _ => Map.empty
```
Comment (fidesachates) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala]: This will match when there are no props, but it's not empty. E.g.

```
{
    "indexnoprops": {
        "mappings": {
            "dynamic": "strict"
        }
    }
}
```

**Comment by fidesachates** [comment]
Sigh. Rethought things. Going to try a third variation.
